CONTRIBUTING.md

```diff
@@ -0,0 +1,125 @@
+# Contributing
+
+Thanks for helping with SlimeVR! This document explains what to know when
+contributing.
+
+We are most active on [discord](https://discord.gg/SlimeVR), come chat with us if you
+have any questions or want to hang out! 😎
+
+
+## Setting up the development environment
+
+We use platformio as our build tooling and rely on git for version control. There are
+instructions to set all of this up [here](https://docs.slimevr.dev/firmware/setup-and-install.html).
+
+
+## Using third-party code
+
+Often you will want to use or reference third party code. You should not just
+copy-paste code from random libraries! There are several reasons for this:
+
+- It is not clear when/where the code originated from.
+- The original code's license needs to be kept intact.
+- It removes the ability to update our code when the original updates.
+- It makes it harder for us to maintain the code, because we don't what has changed
+  from the original!
+
+Instead, we will want to do the following:
+1. Check that the dependency has a permissive license
+2. (Optional) Fork the dependency if it needs any modification
+1. Add the dependency via platformio
+
+
```

Comment:
```suggestion
```

Reply:
This was done intentionally, since there is already a newline between the section header and the section

Reply:
If you format this file with Prettier it will remove these lines (and I think only one line before a heading is the standard).

Reply:
why would we use prettier in a C++ codebase?

Reply:
> why would we use prettier in a C++ codebase?

because it can format markdown.

Reply:
OK we can worry about this if we add a markdown formatter. In the meantime lets get this merged, it should not take a full week to merge an uncontroversial documentation file. Please mark as resolved.